.circleci/config.yml

```
@@ -93,6 +93,75 @@ commands:
             - bytecode-report-<< parameters.label >>-cli.txt
       - matrix_notify_failure_unless_pr
 
+  install_python3:
+    description: "Install python3 and given packages."
+    parameters:
+      packages:
+        description: "List of optional python packages to be installed (separated by space)."
+        type: string
+        default: ""
+    steps:
+      - run:
+          name: Install python and dependencies
+          command: |
+            sudo apt update
+            if ! python3 --version >/dev/null 2>&1
+            then
+              sudo apt install python3 --assume-yes --no-install-recommends
+            fi
+            sudo apt install python3-pip
+            if [ -n "<< parameters.packages >>" ]
+            then
+              echo "Installing additional packages..."
+              python3 -m pip install --user << parameters.packages >>
+            fi
+
+  install_foundry:
+    description: "Install Foundry."
+    parameters:
+      version:
+        type: string
+        default: "nightly"
+      install_path:
+        type: string
+        default: /home/circleci/bin
```

Why not install it system-wide in `/usr/local/bin`?

```suggestion
        default: /usr/local/bin/
```

---

I did like that at first, but as it is an external script and foundry currently does not provide a checksum for their releases, I thought that it would be better to avoid installing things using `sudo`. But just realized now that I left a `sudo` there when extracting the binaries. Just fix it ;)